File: ts-sdk/src/sdk/models/operations/gettermsofserviceuserstatuses.ts
import { SpeakeasyMetadata, SpeakeasyBase, RetryConfig } from "../../../internal/utils";
import * as shared from "../shared";



export class GetTermsOfServiceUserStatusesQueryParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=tos_id" })
  tosId: string;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=user_id" })
  userId?: string;
}


export class GetTermsOfServiceUserStatusesRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  retries?: RetryConfig;

  @SpeakeasyMetadata()
  queryParams: GetTermsOfServiceUserStatusesQueryParams;
}


export class GetTermsOfServiceUserStatusesResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  clientError?: shared.ClientError;

  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  statusCode: number;

  @SpeakeasyMetadata()
  termsOfServiceUserStatuses?: shared.TermsOfServiceUserStatuses;
}

File: sdks/go/pkg/models/operations/gettermsofserviceuserstatuses.go
package operations

import (
	"openapi/pkg/models/shared"
	"openapi/pkg/utils"
)

type GetTermsOfServiceUserStatusesQueryParams struct {
	TosID  string  `queryParam:"style=form,explode=true,name=tos_id"`
	UserID *string `queryParam:"style=form,explode=true,name=user_id"`
}

type GetTermsOfServiceUserStatusesRequest struct {
	Retries     *utils.RetryConfig
	QueryParams GetTermsOfServiceUserStatusesQueryParams
}

type GetTermsOfServiceUserStatusesResponse struct {
	ClientError                *shared.ClientError
	ContentType                string
	StatusCode                 int64
	TermsOfServiceUserStatuses *shared.TermsOfServiceUserStatuses
}
